### PR TITLE
add docstrings to some string files

### PR DESCRIPTION
**Pull request by KristofferC**
More pedagogical examples than I did welcome

### CODE REVIEW

File: base/strings/basic.jl [hunk 1]
```diff
@@ -226,6 +281,18 @@ end
     chr2ind(s::AbstractString, i::Integer)
 
 Convert a character index `i` to a byte index.
+
+See also [`ind2chr`](@ref)
```
Comment (kshyatt) [inline base/strings/basic.jl]: Period at the end of this sentence?

File: base/strings/basic.jl [hunk 2]
```diff
@@ -207,6 +250,18 @@ chr2ind(s::DirectIndexString, i::Integer) = begin checkbounds(s,i); i end
 
 Convert a byte index `i` to a character index with
 respect to string `s`.
+
+See also [`chr2ind`](@ref)
```
Comment (kshyatt) [inline base/strings/basic.jl]: Period at the end?

File: base/strings/util.jl [hunk 3]
```diff
@@ -78,6 +82,10 @@ chop(s::AbstractString) = SubString(s, 1, endof(s)-1)
     chomp(s::AbstractString)
 
 Remove a single trailing newline from a string.
+
+```jldoctest
+julia> chomp("Hello ")
```
Comment (kshyatt) [inline base/strings/util.jl]: Where's the newline here? Or is GitHub just sucking at rendering?

**Review by kshyatt — APPROVED** [review]
Provided the periods get fixed (not urgent!)

**Review by ararslan — APPROVED** [review]

File: base/strings/util.jl [hunk 4]
```diff
@@ -118,6 +126,13 @@ The default delimiters to remove are `' '`, `\\t`, `\\n`, `\\v`,
 `\\f`, and `\\r`.
 If `chars` (a character, or vector or set of characters) is provided,
 instead remove characters contained in it.
+```jldoctest
```
Comment (tkelman) [inline base/strings/util.jl]: most other doctest blocks are set off with an empty line

Comment (KristofferC) [inline base/strings/util.jl]: true, fixed

**Pull request merged**